### PR TITLE
chore: Simplify trivial `async { X.await }` anti-pattern

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

**Review by blt — APPROVED** [review]
Oh, look at that. Nice find.

**Review by pablosichert — APPROVED** [review]
Neat. That looks like a good candidate for a new clippy rule.

**Comment by jszwedko** [comment]
Seems like this would make for a good clippy lint 😄

**Review by jszwedko — APPROVED** [review]

**Review by lukesteensen — APPROVED** [review]

File: src/kubernetes/reflector.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ where
                 let val = select! {
                     // If we got a maintenance request - perform the
                     // maintenance.
-                    _ = maintenance_request.unwrap(), if maintenance_request.is_some() => {
+                    _ = async { maintenance_request.unwrap().await }, if maintenance_request.is_some() => {
```
Comment (jszwedko) [inline src/kubernetes/reflector.rs:115]: I'm curious what you think might be the difference here.

Comment (bruceg) [reply src/kubernetes/reflector.rs:115]: My only idea is that it has to do with whatever the `select!` macro is adding combined with a possible borrow vs ownership change.

**Pull request merged**